setup_unit_tests.py

```diff
@@ -0,0 +1,488 @@
+                output_value_type = list(output_type.members.values())[i]
+            converted_output[output_key] = convert_typed_output_values(output_value, output_value_type, data_dir, get_not_none(metadata, output_value))
+        return converted_output
+    if isinstance(output_type, WDL.Type.Map):
+        converted_output = dict()
+        for output_key, output_value in output_values.items():
+            new_output_key = convert_typed_output_values(output_key, output_type.item_type[0], data_dir, get_not_none(metadata, output_key))
+            new_output_value = convert_typed_output_values(output_value, output_type.item_type[1], data_dir, get_not_none(metadata, output_value))
+            converted_output[new_output_key] = new_output_value
+    if isinstance(output_type, WDL.Type.Pair):
+        converted_output = dict()
+        # key should be left or right
+        converted_output["left"] = convert_typed_output_values(output_values["left"], output_type.left_type, data_dir, get_not_none(metadata, output_values["left"]))
+        converted_output["right"] - convert_typed_output_values(output_values["right"], output_type.right_type, data_dir, get_not_none(metadata, output_values["right"]))
+        return converted_output
+    if isinstance(output_type, WDL.Type.Array):
+        converted_output = list()
+        for i, output in enumerate(output_values):
+            converted_output.append(convert_typed_output_values(output, output_type.item_type, data_dir, index_not_none(metadata, i)))
+        return converted_output
+    # else this is a primitive type
+    return output_values
+
+
+def convert_typed_output_values_from_string(output_values: Union[None, str, Dict[str, Any], List[Any]], output_type: Union[str, Dict[str, Any]],
+                                            data_dir: Optional[Path], metadata: Optional[Dict[str, Any]]):
+    output_type_wdl = convert_type(output_type)
+    return convert_typed_output_values(output_values, output_type_wdl, data_dir, metadata)
+
+
+def recursive_json_apply(json_obj: Union[Dict[Any, Any], List[Any], bool, int, str, float, None], func: Callable[[Any], Any]) \
+        -> Union[Dict[Any, Any], List[Any], bool, int, str, float, None]:
+    if isinstance(json_obj, dict):
+        return {func(k): recursive_json_apply(v, func) for k, v in json_obj.items()}
+    elif isinstance(json_obj, list):
+        return [recursive_json_apply(v, func) for v in json_obj]
+    elif json_obj is None:
+        return None
+    else:
+        # primitive type
+        return func(json_obj)
+
+
+def generate_config_file(m: re.Match, output_dir: Path, version: str, all_data_files: Optional[Set[str]], data_dir: Optional[Path],
+                         output_data_dir: Optional[Path], config: list, metadata: Optional[Dict[str, Any]]) -> None:
+    """
+    Given the regex, create the corresponding config entry for conformance.yaml.
```

Comment:
This isn't a regex, it's a regex match object.

Reply:
Also if this really returns `None`, where is that entry going?

Reply:
> This isn't a regex, it's a regex match object.


Fixed the docstring


> Also if this really returns `None`, where is that entry going?


The entry is appended to the config object which is passed in as an argument. This is the code from the WDL people. I think it's a bit odd to mutate the argument when it can easily be returned, but since it is a minor complain, I probably won't change it.